src/main/java/de/danielbechler/diff/node/DiffNode.java

```diff
@@ -67,6 +67,7 @@
 	private Class<?> valueType;
 	private TypeInfo valueTypeInfo;
 	private IdentityStrategy childIdentityStrategy;
+	private Set<String> categoriesFromConfig;
```

Comment:
Oh, and maybe make it `final` and just initialize it with an empty `Set`. That way we never need to worry about NPEs, which are currently possible because the assiciated setter allows to assign `null`.